examples/run_example.sh

```diff
@@ -58,8 +58,20 @@ on_exit() {
 
 trap on_exit EXIT
 
+if [[ -z "$EXAMPLE" ]]; then
+    fail "$0 requires an example name as argument"
+    info "usage:\n$0 EXAMPLE"
+    exit 1
+fi
+
 SRCDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)" # this source dir
 
+if [[ ! -d "$SRCDIR/$EXAMPLE" || "$EXAMPLE" = "common" ]]; then
```

Comment:
Great idea to include the non-example directories as well, thanks!
The `server` directory should also be excluded.